drivers/flash/Kconfig.stm32

```diff
@@ -13,6 +14,27 @@ config SOC_FLASH_STM32
 	default y
 	select FLASH_PAGE_LAYOUT
 	select FLASH_HAS_PAGE_LAYOUT
+	select FLASH_HAS_EX_OP
```

Comment:
Not an STM32 expert, but shouldn't the `FLASH_HAS_EX_OP` be selected by `SOC_SERIES_STM32F4X` and other SoC specific configs, where the SoC drivers actually provide support for any `ex_op`?

Reply:
Agreed.

Also, I wonder if it would be acceptable to select `FLASH_HAS_EX_OP` based on a device tree input (new compatible/new property). I know this was not acceptable some time ago (SW/HW separation stuff), but maybe this policy has evolved.

Reply:
Like `compatible = "stm32,flash,sector_wp"`?

Reply:
Should be `st,stm32-flash-something`

Reply:
Should I add `st,stm32-flash-ex-op` to existing flash compatible in `dts/arm/st/f4/stm32f4.dtsi` and check it in Kconfig?

Reply:
I am not sure how this would work if we have to add more compatibles per device/node to indicate supported features.

Reply:
Are we talking about compatibility in DTS which indicates that *any* extended operation is supported or about indicating which extended operations are supported?

Support for extended operations (`flash_ex_op()` syscall) can be enabled on all devices supported by the driver (though most devices don't support any extended operation), but WP, RDP, etc. is currently supported on STM32F4 series.

Reply:
I was talking about any extended operation. But I agree this can make things complex if we need to enter details about which operations are supported if not all of them are added at once or if for some reason one is not supported.

So, coming back to the initial point, `select FLASH_HAS_EX_OP` is valid on all series even if no OP is implemented. Is that correct  ?

Reply:
> So, coming back to the initial point, `select FLASH_HAS_EX_OP` is valid on all series even if no OP is implemented. Is that correct ?

Yes. `flash_stm32_ex_op()` will return `-ENOTSUP` if requested OP is not supported.

Reply:
I think that having `select FLASH_HAS_EX_OP` if your driver, or your device, does not actually provide implementation invalidates entire purpose of the Kconfig. The option directly states "this thing has that feature".

Reply:
I can make it `select FLASH_HAS_EX_OP if SOC_SERIES_STM32F4X`

Reply:
> Yeah, but your implementation of `flash_ex_op` can do exactly the same without calling the driver: when none of drivers selects `FLASH_HAS_EX_OP` then the code that goes for flash driver will not even be enabled, shortening call paths to just returning `-ENOTSUP` directly from `FLASH_HAS_EX_OP`.

That's right, but @erwango asked if it's okay to `select FLASH_HAS_EX_OP` even if no extended operations is implemented.

Reply:
> 
> That's right, but @erwango asked if it's okay to `select FLASH_HAS_EX_OP` even if no extended operations are implemented.

IMHO no. Why? For example you have additional part of software that is common for all devices but does additional work when `FLASH_HAS_EX_OP` is enabled (for some reason, comment is for general usage of _HAS_ Kconfig options), now you will have that part of code compiled in, and probably executed, even though your driver does not require it to run but it selects `FLASH_HAS_EX_OP` for no reason.

Reply:
I agree that we should avoid compiling and executing code that effectively does nothing. I thought that the question was about potential problems when `FLASH_HAS_EX_OP` is selected, but no OP is implemented.

Anyway, selecting `FLASH_HAS_EX_OP` only on devices that have any OP implemented is a good thing. The question is if it's okay to do `select FLASH_HAS_EX_OP if SOC_SERIES_STM32F4X` or we should go with DTS solution.

Reply:
Let's go with t's okay to do `select FLASH_HAS_EX_OP if SOC_SERIES_STM32F4X` for now. We can tune things later

Reply:
I did not read all of this discussion, but the official policy on software specific features in DTS is documented here (see "the zephyr, prefix"):

https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html#the-zephyr-prefix

Reply:
Clear, thanks @mbolivar-nordic